opencensus/proto/stats/metrics/metrics.proto

```diff
@@ -69,15 +70,19 @@ message MetricDescriptor {
   Type type = 4;
 
   // TODO: Consider to add LabelInfo{key, description} if needed.
+  // The label keys associated with the metric descriptor. Label keys must be
+  // unique and in sorted order.
+  repeated string label_keys = 5;
 }
 
 // A collection of data points that describes the time-varying values
 // of a metric.
 message TimeSeries {
   // TODO: Add restrictions for characters that can be used for keys and values.
   // The set of label values that uniquely identify this timeseries. Apply to all
-  // points.
-  map<string, string> label = 1;
+  // points. The order of label values must match that of label keys in the
+  // metric descriptor.
+  repeated google.protobuf.StringValue label_values = 1;
```

Comment:
```
// Wrapper message for `string`.
//
// The JSON representation for `StringValue` is JSON string.
message StringValue {
  // The string value.
  string value = 1;
}
```

How does this let us distinguish between empty string and unset?

Reply:
IIUC for empty string we explicitly set `""` for the `value` field of `StringValue`, while for unset we don't set the `value` field (i.e use a default instance).

/cc @bogdandrutu

Reply:
label_value will be null (the whole message) not the string value inside the message.

Reply:
I see, I am not 100% convinced that this whole optimization makes sense with all these complicated things. But here is how I think we can solve this issue:

message LabelValueOrNull {
  // The value for the label.
  string value = 1;
  // If false the value field is ignored and considered not set.
  bool has_value = 2; 
}

Reply:
I think that could go into a separate PR - I put back the TODO about differentiating between empty and unset.

Reply:
+1 for separate PR. Do you want to change `StringValue` back to a regular `string` and drop the `wrappers_proto` dependency?